src/prefect/orion/database/dependencies.py

```diff
@@ -178,11 +183,30 @@ def temporary_orm_config(tmp_orm_config: BaseORMConfiguration):
         MODELS_DEPENDENCIES["orm"] = starting_orm_config
 
 
+@contextmanager
+def temporary_interface_class(tmp_interface_class: BaseORMConfiguration):
+    """
+    Temporarily override the Orion interface class When the context is closed,
+    the existing interface will be restored.
+
+    Args:
+        tmp_interface_class: Orion interface class to inject.
+
+    """
+    starting_interface_class = MODELS_DEPENDENCIES["interface_class"]
+    try:
+        MODELS_DEPENDENCIES["interface_class"] = tmp_interface_class
+        yield
+    finally:
+        MODELS_DEPENDENCIES["interface_class"] = starting_interface_class
```

Comment:
What do you think about:

```suggestion
    try:
        set_interface_class(tmp_interface_class)
        yield
    finally:
        set_interface_class(starting_interface_class)
```

which would let us trigger some additional behavior any time the interface class is set?